nasa/ow_simulator
# Oceanwater 540 factor out ground detection functionality out of trajectory publisher

## Linked Issues:
Linked PR: None
Epic Link: https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-518
Linked Issues:
* https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-540
* https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-527

## Summary of Changes
* Added the ability to track the execution of a trajectory and stop it (implemented in TrajectoryAsyncExecuter)
* Moved out ground detection functionality out of trajectory_publisher
* GuardedMove activity now publishes the result of ground detection as before
* `trajectory_publisher.py` is completely dropped
 
## Test
Launch the simulation
```bash
roslaunch ow (atacama_y1a.launch | europa_terminator.launch | europa_terminator_workspace.launch)
```
In another terminal (with the workspace sourced) use the following command to monitor the output of `/guarded_move_result` topic:
```bash
rostopic echo /guarded_move_result
```
Use the rqt_service_caller interface or using a separate terminal (with the workspace sourced) and invoked any of the ported commands
```bash
rosservice call /arm/guarded_move
```

>Note: Testing this PR through `roslaunch ow_autonomy plan:=ReferenceMission1.plx` wouldn't be possible yet until this PR and https://github.com/nasa/ow_simulator/pull/49 are merged together.

### Test Objectives
* The GuardedMoveResult is published as expected
* No premature or missed ground detection 
* Make sure all other services are performing as expected.
* Check code for any issues or potential improvements.
